samples/resourceops/volume_snapshotop_sequential.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import kfp.dsl as dsl
+
+
+@dsl.pipeline(
+    name="VolumeSnapshotOp Sequential",
+    description="The fourth example of the design doc."
+)
+def volume_snapshotop_sequential(url):
+    vop = dsl.VolumeOp(
+        name="create_volume",
+        resource_name="vol1",
+        size="1Gi",
+        modes=dsl.VOLUME_MODE_RWM
+    )
+
+    step1 = dsl.ContainerOp(
+        name="step1_ingest",
+        image="google/cloud-sdk:216.0.0",
+        command=["sh", "-c"],
+        arguments=["mkdir /data/step1 && "
+                   "gsutil cat %s | gzip -c >/data/step1/file1.gz" % url],
+        volumes={"/data": vop.volume}
+    )
+
+    step1_snap = dsl.VolumeSnapshotOp(
+        name="step1_snap",
+        resource_name="step1_snap",
+        volume=step1.volume
+    )
+
+    step2 = dsl.ContainerOp(
+        name="step2_gunzip",
+        image="library/bash:4.4.23",
+        command=["sh", "-c"],
+        arguments=["mkdir /data/step2 && "
+                   "gunzip /data/step1/file1.gz -c >/data/step2/file1"],
+        volumes={"/data": step1.volume}
```

Comment:
One problem with this approach is that it creates a duplicate slightly different API in ContainerOp, which is confusing.

ContainerOp already has `.volumes` and `.add_volume` and they support Kubernetes volume semantics. This change is replacing an established API with a non-standard mixture of K8s' volume and volumemount.

Can you please implement this using extension methods like the `kfp.onprem.mount_pvc()` does?

It would look like this: `step1 = dsl.ContainerOp(...).apply(mount_volume)` where `mount_volume` is a function or lambda that adds volume and volume mount.